copy.go

```diff
@@ -147,14 +147,14 @@ func Copy(ctx context.Context, src ReadOnlyTarget, srcRef string, dst Target, ds
 	proxy := cas.NewProxyWithLimit(src, cas.NewMemory(), opts.MaxMetadataBytes)
 	root, err := resolveRoot(ctx, src, srcRef, proxy)
 	if err != nil {
-		return ocispec.Descriptor{}, fmt.Errorf("failed to resolve %s: %w", srcRef, err)
+		return ocispec.Descriptor{}, newCopyError("resolveRoot", CopyErrorOriginSource, err)
```

Comment:
"resolveRoot" is internal. It is not that meaningful to end users.

Reply:
refactored